js/bundle/src/decoder.ts

```diff
@@ -31,7 +32,10 @@ export class Bundle {
   constructor(buffer: Uint8Array) {
     const wbn = asArray(cborg.decode(buffer, { useMaps: true }));
 
-    let peekVersion = bytestringToString(wbn[1]).replace(/\0+$/, '');
+    let peekVersion = bytestringToString(wbn[1]).replace(
+      /\0+$/,
+      ''
+    ) as FormatVersion;
```

Comment:
This cast breaks type safety, because the `FormatVersion` type is `"b1" | "b2"` but this value can be arbitrary string.

We should do either:

- Let `FormatVersion` be just `string` (losing some type checks), or
- Validate the value before casting to `FormatVersion`.

Reply:
I added a type predicate to confirm the types!